service/src/main/java/uk/gov/hmcts/reform/fpl/service/cmo/ReviewCMOService.java

```diff
@@ -0,0 +1,119 @@
+package uk.gov.hmcts.reform.fpl.service.cmo;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import lombok.RequiredArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.enums.CMOStatus;
+import uk.gov.hmcts.reform.fpl.exceptions.CMONotFoundException;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+import uk.gov.hmcts.reform.fpl.model.ReviewDecision;
+import uk.gov.hmcts.reform.fpl.model.common.DocumentReference;
+import uk.gov.hmcts.reform.fpl.model.common.Element;
+import uk.gov.hmcts.reform.fpl.model.common.dynamic.DynamicList;
+import uk.gov.hmcts.reform.fpl.model.order.CaseManagementOrder;
+import uk.gov.hmcts.reform.fpl.service.time.Time;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+import static uk.gov.hmcts.reform.fpl.enums.CMOReviewOutcome.JUDGE_AMENDS_DRAFT;
+import static uk.gov.hmcts.reform.fpl.enums.CMOStatus.SEND_TO_JUDGE;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.asDynamicList;
+import static uk.gov.hmcts.reform.fpl.utils.ElementUtils.element;
+
+@Service
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class ReviewCMOService {
+
+    private final ObjectMapper mapper;
+    private final Time time;
+
+    private DynamicList buildDynamicList(List<Element<CaseManagementOrder>> cmos) {
+        return buildDynamicList(cmos, null);
+    }
+
+    public DynamicList buildDynamicList(List<Element<CaseManagementOrder>> cmos, UUID selected) {
```

Comment:
can it accept caseData?
public DynamicList buildReadyForApprovalList(CaseData caseData) {
        List<Element<CaseManagementOrder>> cmosReadyForApproval = this.getCMOsReadyForApproval(
                caseData.getDraftUploadedCMOs());

        Element<CaseManagementOrder> selectedCMO = this.getSelectedCMO(caseData);


        return asDynamicList(cmosReadyForApproval, selectedCMO.getId(), uk.gov.hmcts.reform.fpl.model.order.CaseManagementOrder::getHearing);
    }

It will simplify controller, and make api of this service consistent